ui/src/reusable_ui/components/dropdowns/Dropdown.tsx

```diff
@@ -0,0 +1,194 @@
+// Libraries
+import React, {Component, CSSProperties} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import DropdownDivider from 'src/reusable_ui/components/dropdowns/DropdownDivider'
+import DropdownItem from 'src/reusable_ui/components/dropdowns/DropdownItem'
+import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+
+// Types
+import {
+  ComponentColor,
+  ComponentSize,
+  IconFont,
+  DropdownMenuColors,
+} from 'src/reusable_ui/types'
+
+// Styles
+import './Dropdown.scss'
+
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface Props {
+  children: JSX.Element[]
+  onChange: (value: any) => void
+  selectedItemKey: string
```

Comment:
See later comment about `key`s and `itemKey`s... think this can just be `selectedKey`.